docs/cub/benchmarking.rst

```diff
@@ -1,34 +1,200 @@
 CUB Benchmarks
 *************************************
 
-This file contains instructions on how to run all CUB benchmarks using CUB tuning infrastructure.
+.. TODO(bgruber): this guide applies to Thrust as well. We should rename it to "CCCL Benchmarks" and move it out of CUB
+
+CUB comes with a set of `NVBench <https://github.com/NVIDIA/nvbench>`_-based benchmarks for its algorithms,
+which can be used to measure the performance of CUB on your system on a variety of workloads.
+The integration with NVBench allows to archive and compare benchmark results,
+which is useful for continuous performance testing, detecting regressions, tuning, and optimization.
+This guide gives an introduction into CUB's benchmarking infrastructure.
+
+Building benchmarks
+--------------------------------------------------------------------------------
+
+CUB benchmarks are build as part of the CCCL CMake infrastructure.
+Starting from scratch:
 
 .. code-block:: bash
 
-    pip3 install --user fpzip pandas scipy
     git clone https://github.com/NVIDIA/cccl.git
-    cmake -B build -DCCCL_ENABLE_THRUST=OFF\
-             -DCCCL_ENABLE_LIBCUDACXX=OFF\
-             -DCCCL_ENABLE_CUB=ON\
-             -DCCCL_ENABLE_BENCHMARKS=YES\
-             -DCUB_ENABLE_DIALECT_CPP11=OFF\
-             -DCUB_ENABLE_DIALECT_CPP14=OFF\
-             -DCUB_ENABLE_DIALECT_CPP17=ON\
-             -DCUB_ENABLE_DIALECT_CPP20=OFF\
-             -DCUB_ENABLE_RDC_TESTS=OFF\
-             -DCUB_ENABLE_TUNING=YES\
-             -DCMAKE_BUILD_TYPE=Release\
-             -DCMAKE_CUDA_ARCHITECTURES="89;90"
+    cd cccl
+    mkdir build
     cd build
-    ../cub/benchmarks/scripts/run.py
+    cmake ..\
+        -GNinja\
+        -DCCCL_ENABLE_BENCHMARKS=YES\
+        -DCCCL_ENABLE_CUB=YES\
+        -DCCCL_ENABLE_THRUST=NO\
+        -DCCCL_ENABLE_LIBCUDACXX=NO\
+        -DCUB_ENABLE_RDC_TESTS=NO\
+        -DCMAKE_BUILD_TYPE=Release\
+        -DCMAKE_CUDA_ARCHITECTURES=90 # TODO: Set your GPU architecture
```

Comment:
Rather than using the full fledged matrix we should either add a preset or use a preset and change the few non standard  options. (Suggestion totally untested

```
cmake --preset cub-cxx17 -DCCCL_ENABLE_CUB=YES
```

Reply:
I would like to do this later. Here is an issue: #2839